src/include/sof/audio/buffer.h

```diff
@@ -215,6 +225,77 @@ static inline void *buffer_get_frag(struct comp_buffer *buffer, void *ptr,
 	return current;
 }
 
+/**
+ * Calculates period size in bytes based on component buffer's parameters.
+ * @param buf Component buffer.
+ * @return Period size in bytes.
+ */
+static inline uint32_t buffer_frame_bytes(struct comp_buffer *buf)
+{
+	/* calculate period size based on params */
+	switch (buf->frame_fmt) {
+	case SOF_IPC_FRAME_S16_LE:
+		return 2 * buf->channels;
+	case SOF_IPC_FRAME_S24_4LE:
+	case SOF_IPC_FRAME_S32_LE:
+	case SOF_IPC_FRAME_FLOAT:
+		return 4 * buf->channels;
+	default:
+		return 0;
+	}
```

Comment:
Couldn't this function just do `return buffer_sample_bytes(buf) * buf->channels;` ?

Reply:
I've just rebase PR with master. It's ```return frame_bytes(buf->frame_fmt, buf->channels);``` now.